### DISCUSSION TITLE
Please help this error in wazuh-agent: [ECSIdentity] Agent not taking ECS identity: Could not fetch metadata endpoint

### DISCUSSION THREAD

**Discussion by Truongvt2** [wazuh/wazuh#16921] [General]
Hello everyone,

I setup wazuh to monitor cloudwatchlog in aws, I follow thí guide to install https://github.com/wazuh/wazuh-docker, I chooose singe-node model
From wazuh-agent( install normally by service, not docker-wazuh-agent): I add this configure

```
  <wodle name="aws-s3">
    <disabled>no</disabled>
    <interval>2m</interval>
    <run_on_start>yes</run_on_start>
    <service type="cloudwatchlogs">
      <aws_profile>default</aws_profile>
      <regions>ap-southeast-1</regions>
      <aws_log_groups>aws_ssm_common</aws_log_groups>
    </service>
  </wodle>
```
From wazuh-agent I can run this command normally, and already setup aws credential .aws/config .aws/credential
User-policy of wazuh-user in aws:
 
![policy](https://user-images.githubusercontent.com/32538408/235281591-5443e6c0-b23b-49d3-b9fc-54a3ce226d0e.jpg)

/var/ossec/wodles/aws/aws-s3 --service cloudwatchlogs --aws_profile default --aws_log_groups aws_ssm_common --regions ap-southeast-1 --debug 2

I enable logall_json in osses.conf in wazuh-manager and see this line in archives.log
`full_log: 'DEBUG: The message is "2023-04-28 06:57:22 INFO [IsIdentityEnvironment @ ecs_identity.go.55] [ECSIdentity] Agent not taking ECS identity: Could not fetch metadata endpoint'`

![wazuh](https://user-images.githubusercontent.com/32538408/235281115-8d8008ae-84cd-47f0-b7bf-6d58f3770ac3.png)

Please kindly help to let me know, what's I am wrong , I hope to check log from cloudwatchlog from wazuh

**Comment by Truongvt2** [comment]
hello @Selutario  please kindly help to advise

**Comment by Selutario** [comment]
Hello Truongvt2,

It seems like the SQLite error you reported in [this other discussion](https://github.com/wazuh/wazuh/discussions/13917#discussioncomment-5703976) is now fixed. I see that you are now manually executing the AWS integration:
```
/var/ossec/wodles/aws/aws-s3 --service cloudwatchlogs --aws_profile default --aws_log_groups aws_ssm_common --regions ap-southeast-1 --debug 2
```

That command should be printing any output. Could you share it here after removing any sensitive information? I would like to confirm whether the `[IsIdentityEnvironment @ ecs_identity.go.55] [ECSIdentity] Agent not taking ECS identity: Could not fetch metadata endpoint` message is an error in the Wazuh AWS integration (I could not find any message like that in our code) or if instead, it is a log stored in any of your Cloudwatch `logs streams`, in your AWS account.

In the latter case, Wazuh would be working correctly and would be just finding and fetching the logs stored by Cloudwatch, so this would actually be a log fetched from AWS and not an error.

Note that the AWS Cloudwatch service can store any log message. If enabled, Wazuh will print in both `archives.log` and `archives.json` all logs or events found, but if no decoder or rule matches them, it will not generate alerts that can be indexed and viewed in the dashboard. If this was the case, I would recommend taking a look at this section of the troubleshooting guide:
- [CloudWatch Logs integration is running but no alert is shown in the Wazuh UI - Wazuh docs](https://documentation.wazuh.com/current/amazon/services/troubleshooting.html#cloudwatch-logs-integration-is-running-but-no-alert-is-shown-in-the-wazuh-ui)

**Reply by Truongvt2** [reply]
Hello @Selutario , thank you for your advise. I appreciate your support
After I review my seting again, I see that this error message "[ECSIdentity] Agent not taking ECS identity" from my cloudwatchlog 

![agent_not_taking](https://user-images.githubusercontent.com/32538408/235810050-84f9ce6f-bac2-4061-905c-df52d9e0a22e.jpg)

I will check and fix this error message by my self, please ignore it

Now, I see wazuh-manager already received log from cloudwatchlog
Here is one line when I run command
`root@wazuh:/var/lib/docker/volumes/single-node_wazuh_logs/_data# tail -f archives/archives.json | grep truong`
(Truong is the name, who ssh to ec2 via aws session manager, and i want to logging/parse/show alert of that action in wazuh-dashboard)

![wazuh_no_show](https://user-images.githubusercontent.com/32538408/235811782-52e1160a-a612-4a7a-9d33-56314101b834.jpg)

When I run /var/ossec/bin/wazuh-logtest: I think it's can be parsing  correctly

![image](https://user-images.githubusercontent.com/32538408/235831439-9cd08f15-883c-4ef3-9646-b01be6b33d90.png)

I hope it will be show in wazuh-dashboard, but it's not show

besides, I only see this line in wazuh-dashboard --> Module--> security events

![wazuh123](https://user-images.githubusercontent.com/32538408/235812706-5a8eeeb2-0c1d-4654-8c89-576a9df24518.jpg)

and can not filter in wazuh-dashboard with location=aws-s3

![Screenshot from 2023-05-03 07-30-08](https://user-images.githubusercontent.com/32538408/235812822-be12862a-7c67-4f6c-9972-69e0788a9c76.png)

Please kindly help to advise how to write decode rule to parse above log, to show payload log in wazuh-dashboard

**Reply by Truongvt2** [reply]
hello @Selutario ,just to update
After I run docker-compose down to destroy wazuh and re-install by this guide https://documentation.wazuh.com/current/deployment-options/elastic-stack/all-in-one-deployment/index.html
I installed: filebeat-->elasticsearch-->wazuh-manager-->kibana
and enable archive log in filebeat as recommend of this topic https://groups.google.com/g/wazuh/c/l_TLEkrNw0o
Now I can see cloudwatchlog in dashboard

But I still have one question, I think keywork to make cloudwatchlog show in dashboard is enable archives log and create index patterm : wazuh-archives-4.x*
so if we install wazuh by docker-compose as this guide https://github.com/wazuh/wazuh-docker
where should we enable archives log ? because we dont install filebeat in wazuh-docker

please kindly help to clear

**Reply by Selutario** [reply]
Hello Truongvt2,

Please note that enabling archives implies storing and indexing all logs/events that reach your manager, from all sources, not just those coming from Cloudwatch or that may be relevant enough to generate alerts. This will consume much more space than usual.

It is usually advisable to create rules to match those events that you really consider important. There are multiple guides on how to follow this process:
- [Creating decoders and rules from scratch - Wazuh Blog](https://wazuh.com/blog/creating-decoders-and-rules-from-scratch/)
- [Custom rules and decoders - Wazuh documentation](https://documentation.wazuh.com/current/user-manual/ruleset/custom.html)

If cloudwatch alerts are coming to your dashboard, you can find out how to filter them by displaying the alert details and clicking the symbol below, which will add a filter for that field automatically.
![image](https://user-images.githubusercontent.com/23361101/236131107-0300a651-b962-4ccb-a24c-65b29afbb924.png)

You can also enable the AWS dashboard by following the instructions in the thread you mentioned:
- https://groups.google.com/g/wazuh/c/l_TLEkrNw0o/m/YY-z90eXDQAJ

In case you want to enable archives logging in docker-compose, you should be able to do it in the same way as when you manually installed the components. Accessing the manager container you will be able to see the filebeat configuration file (which is installed by default) under this path:
```
apt-get update
apt-get install vim -y
vi /etc/filebeat/filebeat.yml
```

**Reply by Truongvt2** [reply]
thank you so much @Selutario , I will try to create my rule to filter log from cloudwatch log
